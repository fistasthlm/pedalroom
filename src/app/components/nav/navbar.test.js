import React from 'react';
import { shallow } from 'enzyme';
import Navbar from './navbar';

jest.mock('react-router-dom', () => ({
    withRouter: params => params,
    NavLink: params => params,
}));

const setup = () => {
    return shallow(<Navbar />);
};

describe('Render', () => {
    it('should render component', () => {
        const wrapper = setup();

        expect(wrapper).toMatchSnapshot();
    });
});

describe('State', () => {
    it('should toggleMenu', () => {
        const wrapper = setup();

        wrapper.instance().toggleMenu();

        expect(wrapper.state().menuToggled).toEqual(true);
    });

    it('should close hamburger menu', () => {
        const wrapper = setup();

        wrapper.instance().closeHamburgerMenu();

        expect(wrapper.state().menuToggled).toEqual(false);
    });
});
