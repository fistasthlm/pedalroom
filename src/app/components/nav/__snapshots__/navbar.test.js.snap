// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render should render component 1`] = `
<div
  className="nav-content"
>
  <div
    className="left-div"
    onClick={[Function]}
  >
    <NavLink
      to="/"
    >
      <Image
        className="logo"
        height="80"
        resize={true}
        url="https://images.contentful.com/x1j0zkbk3421/4wgAQ4qPFKIyyeoUImGYko/66256a7ec6c12ea8f8d1d88bbcafe6ea/fistasthlm-logotype.png"
        width="200"
      />
    </NavLink>
  </div>
  <div
    className="right-div"
  >
    <Hamburger
      toggleMenu={[Function]}
    />
  </div>
  <DesktopNavbar
    links={
      Immutable.List [
        Immutable.Map {
          "to": "/",
          "route": "Home",
        },
        Immutable.Map {
          "to": "/bikes",
          "route": "Bikes",
        },
      ]
    }
  />
  <MobileNavbar
    hamburgerMenuStyle="hamburger-menu"
    links={
      Immutable.List [
        Immutable.Map {
          "to": "/",
          "route": "Home",
        },
        Immutable.Map {
          "to": "/bikes",
          "route": "Bikes",
        },
      ]
    }
    toggleMenu={[Function]}
  />
</div>
`
